src/views/TechnicalAssistance/RequestForm/index.tsx

```diff
@@ -0,0 +1,297 @@
+      onClick:
+        idx < step - 1
+          ? e => {
+              history.push(`/trb/requests/${request.id}/${formSteps[idx]}`);
+            }
+          : undefined,
+
+      label: (
+        <>
+          <span className="name">{stp.name}</span>
+          <span className="long">{stp.longName ?? stp.name}</span>
+        </>
+      ),
+      description: stp.description
+    };
+  });
+
+  return (
+    <StepHeader
+      heading={text.heading}
+      text={text.description[0]}
+      subText={text.description[1]}
+      step={step}
+      steps={steps}
+      breadcrumbBar={breadcrumbs}
+    >
+      <UswdsReactLink to="/trb">
+        <IconArrowBack className="margin-right-05 margin-bottom-2px text-tbottom" />
+        {t('button.saveAndExit')}
+      </UswdsReactLink>
+    </StepHeader>
+  );
+}
+
+/**
+ * This is a component base for the TRB request form.
+ * Route param `id` is either `new` or a TrbRequest id.
+ */
+function RequestForm() {
+  const { t } = useTranslation('technicalAssistance');
+
+  const history = useHistory();
+
+  // New requests require `requestType`
+  const location = useLocation<{ requestType: string }>();
+  const requestType = location.state?.requestType as TRBRequestType;
+
+  const { id, step, view } = useParams<{
+    /** Request id */
+    id: string;
+    /** Form step slug */
+    step?: string;
+    /** Form step subview */
+    view?: string;
+  }>();
+
+  const [create, createResult] = useMutation<
+    CreateTrbRequest,
+    CreateTrbRequestVariables
+  >(CreateTrbRequestQuery);
+
+  const [get, getResult] = useLazyQuery<GetTrbRequest, GetTrbRequestVariables>(
+    GetTrbRequestQuery
+  );
+
+  // const [update, updateResult] = useMutation<
+  //   UpdateTrbRequest,
+  //   UpdateTrbRequestVariables
+  // >(UpdateTrbRequestQuery);
+  // todo update, updateResult
+
+  // Assign request data from the first available query response
+  // Prioritize by latest type: update, get, create
+  // eslint-disable-next-line camelcase
+  const request: CreateTrbRequest_createTRBRequest | undefined =
+    getResult.data?.trbRequest || createResult.data?.createTRBRequest;
+  // const data = useMemo(() => {
+  //   if (getResult.data) {
+  //     return getResult.data.trbRequest;
+  //   }
+  //   if (createResult.data) {
+  //     return createResult.data.createTRBRequest;
+  //   }
+  //   return undefined;
+  // }, [getResult, createResult]);
+  // console.log('data', request);
+
+  useEffect(() => {
+    // Create a new request if `id` is new and go to it
+    if (id === 'new') {
+      // Request type must be defined for new requests
+      if (requestType) {
+        if (!createResult.called) {
+          // Create the new request
+          create({ variables: { requestType } }).then(res => {
+            // Update the url with the request id
+            if (res.data) {
+              history.replace(`/trb/requests/${res.data.createTRBRequest.id}`);
```

Comment:
I think this could have stayed as `push`.  If a user creates a request and hits the form, then presses back (either by accident or on purppose) then they have just created a duplicate request.  There may be more than one way to solve for this, either by not allowing to go back, or sending back to homepage, etc.

Reply:
Can you double-check that this is working correctly? I was just testing it and it seems fine. Browser naving back and forth over the creation step doesn't create duplicates, since the `/new` url is replaced and isn't part of history.

Reply:
Hey Adam I tested this and it is creating duplicates when navigating back from `http://localhost:3000/trb/requests/674fa58f-ddd8-4236-bb1e-9adca97baaab/basic` to `http://localhost:3000/trb/steps`.  I think leaving the `.push` would prevent the user from going back once the request is created.  The other redirects can use `.replace`.  Or some other method to prevent the user from `/steps` again.